apps/web-client/src/lib/utils/misc.ts

```diff
@@ -30,3 +30,44 @@ export const extractBookData = <T extends BookData>(entry: T): BookData => ({
 	editedBy: entry.editedBy,
 	outOfPrint: entry.outOfPrint
 });
+
+export function normalizeName(name: string) {
+	// Remove commas and extra spaces, then sort name parts alphabetically
+	return name
+		.replace(/,/g, "") // Remove commas
+		.split(/\s+/) // Split on whitespace
+		.sort() // Sort parts alphabetically
+		.join(" ") // Rejoin with single spaces
+		.trim();
+}
+
+export function matchesName(needle: string, haystack: string) {
+	// Helper to normalize and tokenize the needle string into conditions
+	const getConditions = (str: string) => {
+		return str
+			.toLowerCase()
+			.replace(/[^\w\s]/g, " ") // Replace punctuation with a space
+			.replace(/\s+/g, " ") // Collapse multiple spaces into one
+			.trim()
+			.split(" ")
+			.filter((part) => part.length > 0); // Remove any empty parts resulting from normalization
+	};
```

Comment:
There's no reason for this function to be a closure. It can be moved outside of `matchesName` for increased readability and testability (even if we don't have a test right now, we might always add one in the future if we find a bug with this function).